src/reducers/starWarsReducer.js

```diff
@@ -1,10 +1,33 @@
-import /* we need our action types here*/ "../actions";
+import { GET_SWAPI, ERROR, SUCCESS } from "../actions"; /* we need our action types here*/
 const initialState = {
-  characters: []
+    characters: [],
+    loading: false,
+    error: null
   // Array characters, Boolean fetching, null error.
 };
 export const charsReducer = (state = initialState, action) => {
   switch (action.type) {
+    case GET_SWAPI:
+      return {
+        ...state,
+        loading: true,
+        characters: [],
+        error: null
+      };
+    case SUCCESS:
+      return {
+        ...state,
+        loading: false,
+        characters: action.payload,
+        error: null
+      };
+    case ERROR:
+      return {
+        ...state,
+        loading: false,
+        characters: [],
```

Comment:
again, here you dont need to declare the characters array to an empty array, you can omit this part as it already happens when you ...spread the state in